gitpod-io/website
# [Analytics] Comply with cookie preference for tracking

## Description

Follows the cookie preferences introduced through the new consent flow by not sending analytics events if the User has not agreed to Analytics Tracking. Strictly necessary Segment events (Submission of Feedback, contact forms, messages) are still sent regardless of the preferences set.

## How to test


1. Open a workspace
2. Navigate through the served Website before accepting the cookies
3. Go to the [debugger of Website Staging](https://app.segment.com/gitpod/sources/website_staging/debugger) and ensure that no Analytics events are sent
4. Go to a sales or feedback form, fill in the required data and click send. Make sure that events related to these actions are sent to Segment regardless of the preferences
5. Now accept all on the consent screen and make a few more clicks. Ensure in the debugger that all user events are now tracked

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/2245"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
